### PR TITLE
Change interface of stats query to enable formatting and returning different outputs

### PR DESCRIPTION
**Pull request by kvch**
These changes were required by the upcoming monitoring source.

### CODE REVIEW

File: lib/stats/stats-query-commands.c [hunk 1]
```diff
@@ -33,24 +33,57 @@ typedef enum _QueryCommand
...
+static gboolean
+_ctl_format_get(StatsCluster *sc, StatsCounterItem *ctr, const gchar *ctr_name, gpointer user_data)
+{
+  GString *str = (GString *)user_data;
+  g_string_append_printf(str, "%s.%s: %d\n", sc->query_key, ctr_name ? ctr_name : "", ctr->value);
+  return TRUE;
+}
+
+static gboolean
+_ctl_format_list(StatsCluster *sc, StatsCounterItem *ctr, const gchar *ctr_name, gpointer user_data)
```
Comment (lbudai) [inline lib/stats/stats-query-commands.c]: `_ctl_format_name_without_value` ?

File: lib/stats/stats-query-commands.c [hunk 2]
```diff
@@ -77,31 +110,32 @@ static query_cmd QUERY_CMDS[] =
...
 {
-  GString *result;
+  GString *result = g_string_new("");
   gchar **cmds = g_strsplit(command->str, " ", 3);
 
-  g_assert(g_str_equal(cmds[0], "QUERY"));
+  g_assert(g_str_equal(cmds[CMD_STR], "QUERY"));
 
-  result = _dispatch_query(_command_str_to_id(cmds[1]), cmds[2]);
+  _dispatch_query(_command_str_to_id(cmds[QUERY_CMD_STR]), cmds[QUERY_FILTER_STR], result);
```
Comment (lbudai) [inline lib/stats/stats-query-commands.c]: You are not using the return value.
Will it be used later? :)

File: lib/stats/stats-query.c [hunk 3]
```diff
@@ -42,89 +40,92 @@ static const gchar *_counter_type[SC_TYPE_MAX+1] =
...
-_add_counter_value(StatsCluster *sc, StatsCounterItem *ctr, const gchar *ctr_name, gpointer user_data)
+_add_counter_value(StatsCluster *sc, StatsCounterItem *ctr, const gchar *ctr_name, StatsFormatCb format_cb,
+                   gpointer user_data)
 {
   if (ctr_name && g_str_equal(ctr_name, _counter_type[SC_TYPE_STAMP]))
     return TRUE;
 
-  gint *sum = (gint *) user_data;
+  gpointer *args = (gpointer *) user_data;
+  gchar *sum = (gchar *) args[1];
```
Comment (lbudai) [inline lib/stats/stats-query.c]: gchar * ?

File: lib/stats/stats-query.c [hunk 4]
```diff
@@ -208,63 +209,60 @@ _query_counter_hash(gchar *key_str, gchar *ctr_str, StatsClusterCounterCb wildca
...
 
-GString *
-stats_query_get_sum(const gchar *expr)
+gboolean
+stats_query_get_sum(const gchar *expr, StatsFormatCb format_cb, gpointer result)
 {
   gboolean found_match;
   gchar *key_str = NULL;
   gchar *ctr_str = NULL;
   gint sum = 0;
```
Comment (lbudai) [inline lib/stats/stats-query.c]: maybe gint64?

File: lib/stats/stats-query-commands.c (hunk 2 above)
Comment (kvch) [inline lib/stats/stats-query-commands.c]: It will be used in a later feature on syslog-ng-ctl side.

Comment (lbudai) [inline lib/stats/stats-query-commands.c]: ok

**Pull request merged**